developer_guides/firmware/components/component-overview.rst

```diff
@@ -24,10 +24,14 @@ uAPI to create a component of that ``type``. It also provides an entry point
 to the component ops implementation.
 
 UUIDs (Universally Unique Identifiers) provide a more scalable and
-collision-free way of component identification. UUIDs are currently used by
-the tracing subsystem while both the topology and driver still depend on
-``type`` to define topology and create component instances. UUIDs are
-expected to replace the ``type`` in these parts as well.
+collision-free way of component identification. UUIDs are currently used as
+the standard interface by all users of the SOF firmware, including the
+tracing subsystem, the topology .m4 files, and the Linux topology driver.
+We still support using the ``type`` to define topology and create component
+instances at the moment, but UUIDs are expected to replace the ``type`` in
```

Comment:
It's true, the type will be replaced. per my understanding.

Reply:
it's not my understanding. You need a process/decoder/dai/etc type so that the firmware has a high-level understanding of what the component is.

Reply:
> it's not my understanding. You need a process/decoder/dai/etc type so that the firmware has a high-level understanding of what the component is.

I get your point, if the Linux driver don't need to understand the component type, we can move it out from the struct sof_ipc_comp (e.g. to struct comp_dev), this will allow checking the component type inside firmware such as what comp_get_endpoint_type() do.

Reply:
the PR is updated with this part changed, mark it 'resolved'.